pkgs/desktops/gnome/apps/gnome-text-editor/default.nix

```diff
@@ -0,0 +1,76 @@
+{lib
+, stdenv
+, meson
+, fetchurl
+, python3
+, pkg-config
+, gtk4
+, glib
+, gtksourceview5
+, gsettings-desktop-schemas
+, wrapGAppsHook
+, ninja
+, gnome
+, enchant
+, icu
+, itstool
+, libadwaita
+, libxml2
+, pcre
+, appstream-glib
+, desktop-file-utils
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gnome-text-editor";
+  version = "41.1";
+
+  src = fetchurl {
+    url = "mirror://gnome/sources/gnome-text-editor/${lib.versions.major version}/${pname}-${version}.tar.xz";
+    sha256 = "sha256-YZ7FINbgkF1DEWcCTkPc4Nv2o0Xy1IaTUB1w3HYm+GE=";
+  };
+
+  postPatch = ''
+    chmod +x build-aux/meson/postinstall.py
+    patchShebangs build-aux/meson/postinstall.py
+  '';
+
+  nativeBuildInputs = [
+    appstream-glib
+    desktop-file-utils
```

Comment:
I have been wondering if it is worth it to validate the appstream and desktop files when they have already been validated upstream. Currently, we include `appstream-glib` and `desktop-file-utils` inconsistently so it might be nice to have a policy on that.

Reply:
IMHO it's worth to test them - it takes a second or so to do so and some upstreams don't test for this in CI.